### PR TITLE
feat(IT Wallet): [SIW-1021] Add IT Wallet eID preview screen

### PR DESCRIPTION
**Pull request by mastro993**
> [!WARNING]
> This PR depends on https://github.com/pagopa/io-app/pull/5779, https://github.com/pagopa/io-app/pull/5772

## Short description
This PR adds the eID preview screen with mocked data.

> [!important]
> The screen is missing the component that allows scrolling to the bottom of the page. There is currently an incompatibility with the `IOScrollViewWithLargeHeader` component which will be resolved in a separate PR

## List of changes proposed in this pull request
- Added `ItwIssuanceEidPreviewScreen` component
- Added required locales
- Added mock data for eID, DC, DL and TS in `ItwMockUtils`

## How to test
Navigate to **Profile > IT Wallet > eID > Select an IDP**. You should be able to see the new screen as the below preview.

## Preview

https://github.com/pagopa/io-app/assets/6160324/9d30201f-9c16-4593-8c24-9dab38377389

### CODE REVIEW

File: ts/features/itwallet/issuance/screens/ItwIssuanceEidPreviewScreen.tsx [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+          <VSpacer size={24} />
+          <View style={{ alignSelf: "center" }}>
+            <ButtonLink
+              label={I18n.t(
+                "features.itWallet.issuance.credentialPreview.actions.secondary"
+              )}
+              onPress={() => undefined}
+            />
+          </View>
+        </ContentWrapper>
```
Comment (hevelius) [inline ts/features/itwallet/issuance/screens/ItwIssuanceEidPreviewScreen.tsx:77]: There is a FooterStackButtons (deprecated) that adds footer with two vertically stacked buttons. Btw in the deprecation the hint is to use GradientScrollView. @dmnplb is this the right solution to get vertically stacked buttons?

File: ts/features/itwallet/issuance/screens/ItwIssuanceEidPreviewScreen.tsx [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+    backgroundColor: IOColors.white,
+    shadowColor: IOColors.black,
+    shadowOffset: { width: 0, height: -4 },
+    shadowRadius: 20,
+    shadowOpacity: 0.2,
+    elevation: 5
+  },
+  content: {
+    flex: 1,
+    backgroundColor: IOColors.white, // Add background to cover shadow
```
Comment (hevelius) [inline ts/features/itwallet/issuance/screens/ItwIssuanceEidPreviewScreen.tsx:114]: nice trick :-)

File: ts/features/itwallet/issuance/screens/ItwIssuanceEidPreviewScreen.tsx (hunk 1 above)
Comment (mastro993) [reply ts/features/itwallet/issuance/screens/ItwIssuanceEidPreviewScreen.tsx:77]: We could wait for the merge of this https://github.com/pagopa/io-app/pull/5787

Comment (dmnplb) [reply ts/features/itwallet/issuance/screens/ItwIssuanceEidPreviewScreen.tsx:77]: @hevelius @mastro993 Better use the new `FooterActions` in this case (which is visually similar to the old `FooterStackButtons`)

File: ts/features/itwallet/issuance/screens/ItwIssuanceEidPreviewScreen.tsx (hunk 2 above)
Comment (dmnplb) [reply ts/features/itwallet/issuance/screens/ItwIssuanceEidPreviewScreen.tsx:114]: @mastro993 What about using the `appBackground-primary` theme key, instead? Yes, I know the dark mode is not our top priority, but I'm a hopeless optimist

Comment (mastro993) [reply ts/features/itwallet/issuance/screens/ItwIssuanceEidPreviewScreen.tsx]: > @mastro993 What about using the `appBackground-primary` theme key, instead? Yes, I know the dark mode is not our top priority, but I'm a hopeless optimist

Added in https://github.com/pagopa/io-app/pull/5780/commits/7f4c5942cfd7beb05b55d7c5f4412e5631c92c5a

File: ts/features/itwallet/issuance/screens/ItwIssuanceEidPreviewScreen.tsx (hunk 1 above)
Comment (mastro993) [reply ts/features/itwallet/issuance/screens/ItwIssuanceEidPreviewScreen.tsx:83]: > @hevelius @mastro993 Better use the new `FooterActions` in this case (which is visually similar to the old `FooterStackButtons`)

`FooterActions` does not work properly with the `IOScrollViewWithLargeHeader`. I think it's better to merge this PR and then think about a proper solution later.

**Review by hevelius — APPROVED** [review]
LGTM. Related issues with Footer could be solved in a next PR.

File: ts/features/itwallet/issuance/screens/ItwIssuanceEidPreviewScreen.tsx [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+          <VSpacer size={24} />
+          <View style={{ alignSelf: "center" }}>
+            <ButtonLink
+              label={I18n.t(
+                "features.itWallet.issuance.credentialPreview.actions.secondary"
+              )}
+              onPress={() => undefined}
+            />
+          </View>
+        </ContentWrapper>
```
Comment (dmnplb) [inline ts/features/itwallet/issuance/screens/ItwIssuanceEidPreviewScreen.tsx:83]: Just replaced this chunk of code but as @mastro993 pointed out, an extra margin was applied at the end of the page. I've already changed `IOScrollView` and `IOScrollViewWithLargeHeader` with an extra prop called `excludeEndContentMargin`. Let me know if I can add the change to this PR or if you prefer to manage it in a different PR
```suggestion
<FooterActions
          fixed={false}
          actions={{
            type: "TwoButtons",
            primary: {
              icon: "add",
              iconPosition: "end",
              label: I18n.t(
                "features.itWallet.issuance.credentialPreview.actions.primary"
              ),
              onPress: () => undefined
            },
            secondary: {
              label: I18n.t(
                "features.itWallet.issuance.credentialPreview.actions.secondary"
              ),
              onPress: () => undefined
            }
          }}
        />
```

Comment (mastro993) [reply ts/features/itwallet/issuance/screens/ItwIssuanceEidPreviewScreen.tsx:83]: > Just replaced this chunk of code but as @mastro993 pointed out, an extra margin was applied at the end of the page. I've already changed `IOScrollView` and `IOScrollViewWithLargeHeader` with an extra prop called `excludeEndContentMargin`. Let me know if I can add the change to this PR or if you prefer to manage it in a different PR

If we already have the solution, let's add it to this PR! 🚀

Comment (dmnplb) [reply ts/features/itwallet/issuance/screens/ItwIssuanceEidPreviewScreen.tsx]: @mastro993 Addressed in the following commit → https://github.com/pagopa/io-app/pull/5780/commits/874a314daca5c6755f7eb230bed66ff73df0984b. Let me know if everything is okay ✌🏻 

(Tested on iPhone SE, first generation one, and here's how that turned out)

Comment (mastro993) [reply ts/features/itwallet/issuance/screens/ItwIssuanceEidPreviewScreen.tsx]: It works as expected. Thank you!

**Pull request merged**